prod/mach_zehnder_gadget/javascript/app/WebSocketController.js

```diff
@@ -10,29 +9,377 @@ myApp.controller('WebSocketController',['$scope', function($scope) {
+
+        if (!bs2IsActivated) {
+            var newPoints3 = scalePoints(horizontalBottom, scale);
+            if (!filterIsActivated) 
+                var newPoints8 = scalePoints(tri, scale);
+            else
+                var newPoints8 = scalePoints(triQuant, scale);
+        }
+        else {
+            var newPoints3 = scalePoints(horizontalBottom2, scale);
+            var newPoints8 = scalePoints(empty, scale);
+        }
+
+        if (bs1IsActivated && bs2IsActivated)
+            $(".straight-line-quant").attr("opacity", 0);
+        else
+            if (bs1IsActivated || bs2IsActivated)
+                $(".straight-line-quant").attr("opacity", 0.3);
+            else
+                $(".straight-line-quant").attr("opacity", 0.6);
+
+        firstPath.attr("d", line(newPoints));
+        secondPath.attr("d", line(newPoints6));
+        horizontalTopPath.attr("d", line(newPoints1));
+        verticalLeftPath.attr("d", line(newPoints2));
+        hortizontalBottomPath.attr("d", line(newPoints3));
+        verticalRightPath.attr("d", line(newPoints4));
+        triRightPath.attr("d", line(newPoints7));
+        triBottomPath.attr("d", line(newPoints8));
+        quantPath.attr("d", line(newPoints9));
+        lastPath.attr("d", line(newPoints5));
+        $('#vis').css({'height': 460 * scale, 'width': 860 * scale});
+        $('#diagramDiv').css({'height': 500 * scale, 'width': 860 * scale});
+        $('#saveButton').css({'font-size': (scale * 16) + 'px'})
+        var videoWidth = $('#video1').width();
+        $('#video1, #video2, #mycanvas, #mycanvas2').css('height', videoWidth*0.667);
+
+        if (chart) {
+            chart.options.height = videoWidth*0.667;
+            chart.render();
+        }
+    }
+
+    function scalePoints (points, scale) {
+      var newPoints = [];
+
+      for (var i = 0; i < points.length; i++) {
+        newPoints[i] = {x: points[i].x * scale, y: points[i].y * scale};
+      }
+
+      return newPoints;
+    } 
+
+    function x(d) { return d.x; }
+    function y(d) { return d.y; }
+
+    $(window).on('resize', resize);
+
+    var initializeSocket = function() {
+        ws = new WebSocket('ws://' + host + ':' + port);
+        ws.onopen = function() { 
+            connected = 1;
+            var sensorRequest = {
+                method: 'getSensorData',
+                sensorId: 'INFO_S',
+                accessRole: 'controller'
+            }
+            jsonRequest = JSON.stringify(sensorRequest);
+            if (connected)
+                ws.send(jsonRequest);
+
+            var actuatorRequest = {
+                authToken: 'skfjs343kjKJ',
+                method: 'sendActuatorData',
+                accessRole: 'controller',
+                actuatorId: 'power',
+                valueNames: "",
+                data: [0]
+            };
+            var jsonRequest = JSON.stringify(actuatorRequest);
+            if (connected)
+                ws.send(jsonRequest);
+
```

Comment:
be consistent in leaving or omitting empty lines between case structures and rest of code